src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+                AgentVersion = "peerB",
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            await swarmB.StartAsync();
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/0.0.0.0/tcp/0");
+            var peerBPort = peerBAddress.Protocols[1].Value;
+            Assert.IsTrue(peerB.Addresses.Count() > 0);
+
+            var swarm = new SwarmService {LocalPeer = self};
+            await swarm.StartAsync();
+            try
+            {
+                MultiAddress ma = $"/ip4/127.0.0.100/tcp/{peerBPort}/ipfs/{peerB.Id}";
+                var connection = await swarm.ConnectAsync(ma);
+                var remote = connection.RemotePeer;
+                Assert.AreEqual(remote.ConnectedAddress, ma);
+                CollectionAssert.Contains(remote.Addresses.ToArray(), ma);
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task Connect_CancelsOnStop()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var venus = new Peer
+            {
+                Id = "QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+                Addresses = new MultiAddress[]
+                {
+                    "/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ", // mars.i.ipfs.io
+                }
+            };
+
+            await swarm.StartAsync();
+            var a = swarm.ConnectAsync(venus);
+            Assert.IsFalse(a.IsCanceled || a.IsFaulted);
+
+            await swarm.StopAsync();
+            var endTime = DateTime.Now.AddSeconds(3);
+            while (!a.IsCanceled && !a.IsFaulted)
+            {
+                if (DateTime.Now > endTime)
+                    Assert.Fail("swarm did not cancel pending connection.");
+                await Task.Delay(100);
+            }
+
+            Assert.IsTrue(a.IsCanceled || a.IsFaulted);
+        }
+
+        [TestMethod]
+        public async Task Connect_IsPending()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var venus = new Peer
+            {
+                Id = "QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+                Addresses = new MultiAddress[]
+                {
+                    "/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ", // mars.i.ipfs.io
+                }
+            };
+
+            await swarm.StartAsync();
+            try
+            {
+                Assert.IsFalse(swarm.HasPendingConnection(venus));
+
+                var a = swarm.ConnectAsync(venus);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'a'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)